sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4603,6 +4603,14 @@ object SQLConf {
     .booleanConf
     .createWithDefault(true)
 
+  val LEGACY_NO_CHAR_PADDING_IN_PREDICATE = buildConf("spark.sql.legacy.noCharPaddingInPredicate")
+    .internal()
+    .doc("When true, Spark will not apply char type padding for CHAR type columns in string " +
+      s"comparison predicates, when '${READ_SIDE_CHAR_PADDING.key}' is false.")
+    .version("4.0.0")
```

Comment:
QQ: Is it needed by 3.5 or earlier?

Reply:
We met [SPARK-48562](https://issues.apache.org/jira/browse/SPARK-48562) in Spark 3.5, which is a bug for `ApplyCharTypePadding` while writing JDBC temporary views, it lead us turn off READ_SIDE_CHAR_PADDING while creating JDBC temp views. Thus, we need backport this pr or filx the [SPARK-48562](https://issues.apache.org/jira/browse/SPARK-48562) in 3.5.

Reply:
I see. Feel free to create a backport PR and I can help merge it.